sql/src/main/java/io/crate/analyze/ResultSetStatement.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.analyze;
+
+import io.crate.expression.symbol.Field;
+
+import javax.annotation.Nullable;
+import java.util.List;
+
+public interface ResultSetStatement extends AnalyzedStatement {
+
+    @Nullable
```

Comment:
I think this should not be `nullable`. This seems to be the case only because in the UPDATE case RETURNING is optional. Which makes me wonder if having a separate interface/doing instance of checks on it is the right approach.

Maybe `AnalyzedStatement` generally has nullable fields?